envoy/scripts/copy_envoy_config.sh

```diff
@@ -15,16 +15,27 @@ if [ -f "$ENVOY_CONFIG_PATH" ]; then
     exit 1
 fi
 
-# Prompt for AUTH_DOMAIN
-read -p "Enter AUTH_DOMAIN (eg. 'auth.example.com'): " AUTH_DOMAIN
-
-# Prompt for AUTH_AUDIENCE
-read -p "Enter AUTH_AUDIENCE (eg. 'https://auth.example.com/oauth/token'): " AUTH_AUDIENCE
-
-# Substitute sensitive variables manually using bash parameter expansion
-sed -e "s|\${AUTH_DOMAIN}|$AUTH_DOMAIN|g" \
-    -e "s|\${AUTH_AUDIENCE}|$AUTH_AUDIENCE|g" \
-    "$SCRIPT_DIR/../envoy.template.yaml" > "$ENVOY_CONFIG_PATH"
+# Prompt the user if they wish to use an OAuth provider
+read -p "Do you wish to use an OAuth provider to enable authorizing requests using JWT? (yes/no): " USE_OAUTH
```

Comment:
@commoddity Can you make `no` the default please?

Reply:
There's no default; it requires the user to answer `yes|y`or `no|n`.

Reply:
> @commoddity Can you make `no` the default please?

I've improved the prompt to read better; it now takes 1. and 2. as options:

```bash
Configure JWT Authentication:
1. Enable JWT Auth (requires an OAuth provider like Auth0, along with domain and audience details)
2. Disable JWT Auth
Select an option (1 or 2): 
> 
```

I've also updated the script to use `yq` to remove the JWT-specific parts of the `.envoy.yaml` file if the user does not wish to use JWT as those parts will interfere with the proper functioning of the Envoy proxy unless the user has an 0Auth provider configured. (This was a TODO that I figured I should just do now.)